spec/compiler/semantic/while_spec.cr

```diff
@@ -6,11 +6,37 @@ describe "Semantic: while" do
   end
 
   it "types while with break without value" do
-    assert_type("while true; break; end") { nil_type }
+    assert_type("while 1; break; end") { nil_type }
```

Comment:
What's the difference between using 1 and true here? (I see there's also a spec below with true, and the type ends up being the same). Just curious.

Reply:
The compiler doesn't handle always-truthy types at this point, so the two types are formed differently; this one is `Nil | Nil` (coming from the expression and the break), the `while true` one is exactly `Nil`.